front/pages/api/w/[wId]/workspace-usage.ts

```diff
@@ -15,21 +21,42 @@ const MonthSchema = t.refinement(
   "YYYY-MM"
 );
 
+export const usageTables = [
+  "users",
+  "assistant_messages",
+  "builders",
+  "agent_configurations",
+  "all",
+];
+type usageTableType = (typeof usageTables)[number];
+
+export function getSupportedUsageTablesCodec(): t.Mixed {
```

Comment:
Do you mind adding a comment explaining what this does ? I'm actually quite surprised this is working at all, are you getting any type inference on the decoded values ?

Reply:
@fontanierh Unfortunately no I am not getting any type here, `query.table` is of type `any`. 
We are doing this in a bunch of places to dynamically create the union type:
- https://github.com/dust-tt/dust/blob/ed7bc31e9e8f3c19ddc4090011454c30368bb11a/front/lib/api/pagination.ts#L24
- https://github.com/dust-tt/dust/blob/ed7bc31e9e8f3c19ddc4090011454c30368bb11a/types/src/front/content_fragment.ts#L52

Would you recommend doing it differently?

Reply:
I don't know if there's a better way TBH. Don't want to block for this if we're doing it elsewhere